crates/router/src/connector/paypal/transformers.rs

```diff
@@ -0,0 +1,540 @@
+                connector_metadata: Some(
+                    serde_json::to_value(PaypalMeta { txn_id: id }).unwrap_or_default(),
+                ),
+            }),
+            ..item.data
+        })
+    }
+}
+
+#[derive(Debug, Serialize)]
+pub struct PaypalPaymentsCaptureRequest {
+    amount: OrderAmount,
+    final_capture: bool,
+}
+
+impl TryFrom<&types::PaymentsCaptureRouterData> for PaypalPaymentsCaptureRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsCaptureRouterData) -> Result<Self, Self::Error> {
+        let amount = OrderAmount {
+            currency_code: item.request.currency,
+            value: item.request.amount.to_string(),
+        };
+        Ok(Self {
+            amount,
+            final_capture: true,
+        })
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum PaypalCaptureStatus {
+    Completed,
+    Declined,
+    Refunded,
+    Failed,
+    Pending,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PaymentCaptureResponse {
+    id: String,
+    status: PaypalCaptureStatus,
+    amount: Option<OrderAmount>,
+    final_capture: bool,
+}
+
+impl TryFrom<types::PaymentsCaptureResponseRouterData<PaymentCaptureResponse>>
+    for types::PaymentsCaptureRouterData
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::PaymentsCaptureResponseRouterData<PaymentCaptureResponse>,
+    ) -> Result<Self, Self::Error> {
+        let amount_captured = item.data.request.amount_to_capture;
+        let status = match item.response.status {
+            PaypalCaptureStatus::Completed => storage_enums::AttemptStatus::Charged,
+            PaypalCaptureStatus::Declined => storage_enums::AttemptStatus::Failure,
+            PaypalCaptureStatus::Failed => storage_enums::AttemptStatus::CaptureFailed,
+            PaypalCaptureStatus::Refunded => storage_enums::AttemptStatus::AutoRefunded,
+            PaypalCaptureStatus::Pending => storage_enums::AttemptStatus::Pending,
+        };
+        Ok(Self {
+            status,
+            response: Ok(types::PaymentsResponseData::TransactionResponse {
+                resource_id: types::ResponseId::ConnectorTransactionId(
+                    item.data.request.connector_transaction_id.clone(),
+                ),
+                redirection_data: None,
+                mandate_reference: None,
+                connector_metadata: Some(
+                    serde_json::to_value(PaypalMeta {
+                        txn_id: item.response.id,
+                    })
+                    .unwrap_or_default(),
+                ),
+            }),
+            amount_captured,
+            ..item.data
+        })
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum PaypalCancelStatus {
```

Comment:
Why do we need separate enum here? We can make use of `PaypalCaptureStatus ` right?

Reply:
As we discussed, this need separate enum